src/App.test.js

```diff
@@ -9,85 +8,61 @@ import App from './App';
 import { IntlProvider } from 'react-intl';
 import messages_es from './i18n/es.json';
 import messages_en from './i18n/en.json';
-import jwt_decode from 'jwt-decode';
-
-//Add mock to decode jwt token and not fail
-jest.mock('jwt-decode');
 
 const messages = {
   es: messages_es,
   en: messages_en,
 };
 
-const response = {
-  data: {
-    user: {
-      Email: 'fcoronel@makingsense.com',
+function createDoubleSessionManager() {
+  const double = {
+    initialize: (handler) => {
+      double.updateAppSession = handler;
     },
-  },
-};
-
-const responseToken = {
-  data: {
-    jwtToken: 'token',
-  },
-};
-
-//Local storage mock
-const localStorageMock = {
-  getItem: jest.fn(),
-  setItem: jest.fn(),
-  removeItem: jest.fn(),
-};
-global.localStorage = localStorageMock;
+    finalize: () => {},
+    session: {
+      status: 'non-authenticated',
+    },
+  };
+  return double;
+}
 
 describe('App component', () => {
   afterEach(cleanup);
 
-  beforeEach(() => {
-    axios.get = jest.fn((url) => {
-      if (url === process.env.REACT_APP_API_URL + '/Reports/Reports/GetUserData') {
-        return Promise.resolve(response);
-      } else {
-        return Promise.resolve(responseToken);
-      }
-    });
-  });
-
   it('renders app component', () => {
+    const dependencies = {
+      sessionManager: createDoubleSessionManager(),
+    };
+
     const { getByText } = render(
       <IntlProvider locale="en" messages={flattenMessages(messages['en'])}>
-        <App />
+        <App dependencies={dependencies} />
       </IntlProvider>,
     );
   });
 
-  it('fetches user and display user data', async () => {
-    const tokenDecodeData = {
-      email: 'fcoronel@makingsense',
-      name: 'fede',
-      lang: 'es',
-    };
+  it('updates content after successful authentication', async () => {
+    // Arrange
+    const expectedEmail = 'fcoronel@makingsense.com';
 
-    jwt_decode.mockResolvedValue(tokenDecodeData);
+    const dependencies = {
+      sessionManager: createDoubleSessionManager(),
+    };
 
     const { getByText } = render(
       <IntlProvider locale="en" messages={flattenMessages(messages['en'])}>
-        <App />
+        <App dependencies={dependencies} />
       </IntlProvider>,
     );
 
-    await wait(() => getByText(response.data.user.Email));
+    getByText('Loading...');
 
-    const userEmail = getByText(response.data.user.Email);
+    // Act
+    dependencies.sessionManager.updateAppSession({ status: 'authenticated' });
 
-    expect(userEmail).toBeDefined();
-    expect(axios.get).toHaveBeenCalledTimes(1);
-    expect(axios.get).toHaveBeenCalledWith(
-      process.env.REACT_APP_API_URL + '/Reports/Reports/GetUserData',
-      {
-        withCredentials: 'include',
-      },
-    );
+    // Assert
+    getByText(expectedEmail);
```

Comment:
this work?

Reply:
I did not understand your question.

Reply:
cuando pasas el ` const expectedEmail = 'fcoronel@makingsense.com';` al componente? eso no veia

Reply:
Está hardcodeada la respuesta. Tengo varias mejoras pendientes con los test, una de ellas tiene que ver con esto, disculpá la desprolijidad :(